drivers/serial/uart_altera_jtag.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <drivers/uart.h>
+#include <sys/sys_io.h>
+
+#define DT_DRV_COMPAT   altr_jtag_uart
+
+#define UART_ALTERA_JTAG_DATA_OFFSET    0x00        /* DATA : Register offset */
+#define UART_ALTERA_JTAG_CTRL_OFFSET    0x04        /* CTRL : Register offset */
+#define UART_IE_TX                      (1 << 0)    /* CTRL : TX Interrupt Enable */
+#define UART_IE_RX                      (1 << 1)    /* CTRL : RX Interrupt Enable */
+#define UART_DATA_MASK                  0xFF        /* DATA : Data Mask */
+#define UART_WFIFO_MASK                 0xFFFF0000  /* CTRL : Transmit FIFO Mask */
+
+#ifdef  CONFIG_UART_ALTERA_JTAG_HAL
+#include "altera_avalon_jtag_uart.h"
+#include "altera_avalon_jtag_uart_regs.h"
+
+extern int altera_avalon_jtag_uart_read(altera_avalon_jtag_uart_state *sp,
+		char *buffer, int space, int flags);
+extern int altera_avalon_jtag_uart_write(altera_avalon_jtag_uart_state *sp,
+		const char *ptr, int count, int flags);
+#else
+
+/* device data */
+struct uart_altera_jtag_device_data {
+    struct k_spinlock lock;
+};
+
+/* device config */
+struct uart_altera_jtag_device_config {
+    mm_reg_t base;
+};
+#endif /* CONFIG_UART_ALTERA_JTAG_HAL */
+
+static void uart_altera_jtag_poll_out(const struct device *dev,
+					       unsigned char c)
+{
+#ifdef  CONFIG_UART_ALTERA_JTAG_HAL
+	altera_avalon_jtag_uart_state ustate;
+
+	ustate.base = JTAG_UART_0_BASE;
+	altera_avalon_jtag_uart_write(&ustate, &c, 1, 0);
+#else
+    const struct uart_altera_jtag_device_config *config = dev->config;
+    struct uart_altera_jtag_device_data *data = dev->data;
+
+    k_spinlock_key_t key = k_spin_lock(&data->lock);
+
+    /* While TX FIFO full */
+    while(!(sys_read32(config->base + UART_ALTERA_JTAG_CTRL_OFFSET) & UART_WFIFO_MASK)){
+    }
+
+    uint32_t data_val = sys_read32(config->base + UART_ALTERA_JTAG_DATA_OFFSET);
+    data_val &= ~UART_DATA_MASK;
+    data_val |= c;
+    sys_write32(data_val, config->base + UART_ALTERA_JTAG_DATA_OFFSET);
+
+    k_spin_unlock(&data->lock, key);
```

Comment:
Please align the indentation in this file.

Reply:
I had removed the merge commit and clean up the indentation errors.

However, there is 1 indentation error that I couldn't resolve.

c8c393e2e3fdf5d28ae45c4bba845e9b18e0e022:78: ERROR:SPACING: need consistent spacing around '*' (ctx:WxV)
#78: FILE: drivers/serial/uart_altera_jtag.c:26:
+extern int altera_avalon_jtag_uart_read(altera_avalon_jtag_uart_state *sp,
                                                                       ^
c8c393e2e3fdf5d28ae45c4bba845e9b18e0e022 total: 1 errors, 0 warnings, 143 lines checked

It seems like it misinterpret *pointer to multiplication. Any suggestion what I can do?